[DNM] sr: Relax reference lookup

michael-redpanda opened the pull request:
New schemas may now use soft-deleted subjects as references.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [X] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

* none

michael-redpanda committed:
sr: Relax reference lookup

New schemas may now use soft-deleted subjects as references.

Signed-off-by: Michael Boquard <michael@redpanda.com>

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--aae9242b-db68-4935-b053-10d522c2756d-->
<!--TEST RESULTS-->
<!--a313b8a1-3c28-45f2-95dc-46d1b310c4c7-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/64172">build#64172</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.S3.retention_type=retention.bytes | ducktape | https://buildkite.com/redpanda/redpanda/builds/64172#0195fdb4-3ba8-49b4-b8ed-adb64dcca1c8 | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/64172#0195fdb8-d3b6-4f19-a54c-3954875964aa | FLAKY | 1/3 |
| rptest.tests.controller_log_limiting_test.ControllerLogLimitMirrorMakerTests.test_mirror_maker_with_limits | ducktape | https://buildkite.com/redpanda/redpanda/builds/64172#0195fdb8-d3b7-4bc9-97e0-6ef7f480423f | FLAKY | 1/2 |
| rptest.tests.datalake.compaction_gaps_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/64172#0195fdb8-d3b7-4559-95dc-43d54765c1e7 | FLAKY | 1/5 |
| storage_single_thread_rpunit.storage_single_thread_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/64172#0195fd59-4c06-495c-a588-0b01793e1a1a | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/64194">build#64194</a></summary>


| test_id                                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   |
|:------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery                            | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64194#01960181-e63d-43e9-a678-314eafa53abd | FLAKY         | 1/2      |
| rptest.tests.schema_registry_test.SchemaRegistryAutoAuthTest.test_imported_schemas_with_dependencies_issues | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64194#01960181-e63e-48c2-acf3-64fe5101b9f3 | FAIL          | 0/20     |
| rptest.tests.schema_registry_test.SchemaRegistryAutoAuthTest.test_imported_schemas_with_dependencies_issues | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64194#01960185-a27b-4139-93b7-25504529ff25 | FAIL          | 0/20     |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_imported_schemas_with_dependencies_issues         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64194#01960181-e63e-48c2-acf3-64fe5101b9f3 | FAIL          | 0/20     |
| rptest.tests.schema_registry_test.SchemaRegistryTest.test_imported_schemas_with_dependencies_issues         | ducktape    | https://buildkite.com/redpanda/redpanda/builds/64194#01960185-a27b-4139-93b7-25504529ff25 | FAIL          | 0/20     |
</details>
<!--/body-->

michael-redpanda commented:
/test-release-pipeline

michael-redpanda committed:
sr: Permit returning invalid schemas

Bit of a hack around get_subject_schema to return a schema that is
technically invalid.  Will still throw if the subject is not found.

Signed-off-by: Michael Boquard <michael@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/64194-->
<!--6a1f92a8-6411-4534-b767-3f6ef0115ffa-->
<!--69ff40a2-b614-444d-861c-57a82debbbe5-->
<!--title-->
## Retry command for [Build#64194](https://buildkite.com/redpanda/redpanda/builds/64194)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```

/ci-repeat 1
tests/rptest/tests/schema_registry_test.py::SchemaRegistryAutoAuthTest.test_imported_schemas_with_dependencies_issues
tests/rptest/tests/schema_registry_test.py::SchemaRegistryTest.test_imported_schemas_with_dependencies_issues
```
<!--/body-->

michael-redpanda commented:
Above test failures are expected as these schemas were originally not able to be seen but thanks to the changes in this PR, they can be seen now.  Will need to fix them up later

michael-redpanda commented:
/test-release-pipeline
